src/Client.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+const fs = require('fs');
+const tls = require('tls');
+const os = require('os');
+const PulsarBinding = require('./pulsar-binding');
+
+const certsFilePath = `${__dirname}/cert.pem`;
+
+class Client {
+  constructor(params) {
+    if (!params.tlsTrustCertsFilePath) {
+      // eslint-disable-next-line no-param-reassign
+      params.tlsTrustCertsFilePath = certsFilePath;
+    }
+    this.client = new PulsarBinding.Client(params);
+  }
+
+  createProducer(params) {
+    return this.client.createProducer(params);
+  }
+
+  subscribe(params) {
+    return this.client.subscribe(params);
+  }
+
+  createReader(params) {
+    return this.client.createReader(params);
+  }
+
+  close() {
+    this.client.close();
+  }
+
+  static setLogHandler(params) {
+    PulsarBinding.Client.setLogHandler(params);
+  }
+
+  static genCertFile() {
+    fs.rmSync(certsFilePath, { force: true });
```

Comment:
What happens if you bootstrap two clients on the same machine?
Will they use the same file?

Also, we need to set permissions appropriately on this file, otherwise other users may override the contents and lead to a security hole

Cc @michaeljmarshall @lhotari @BewareMyPower

Reply:
> What happens if you bootstrap two clients on the same machine?
Will they use the same file?

The file is generated only once at installation time. 
If pulsar-client is the global install, they will use the same file.
If pulsar-client is not installed globally, each project generates files and uses them separately.

> Also, we need to set permissions appropriately on this file, otherwise other users may override the contents and lead to a security hole

How do I set permissions? I don't think it's necessary, just like the default `/etc/ssl/cert.pem` file in linux, users can override it too, right?

Reply:
The permission of `ca-certificates.crt` on Linux is:

```bash
$ ls -lh /etc/ssl/certs/ca-certificates.crt
-rw-r--r-- 1 root root 191K Feb  7 15:20 /etc/ssl/certs/ca-certificates.crt
```

But I don't think we need to change the permission. Because this file would be installed under `node_modules/` in the project directory. The owner is who ran `npm install`. It would be that user's fault if the permission was overridden by others.